packages/svelte-ux/src/routes/docs/components/Dialog/+page.svelte

```diff
@@ -188,8 +191,43 @@
     <Dialog {open} on:close={toggleOff} persistent>
       <div slot="title">Are you sure you want to do that?</div>
       <div slot="actions">
-        <Button variant="fill" color="primary">Yes</Button>
-        <Button>No</Button>
+        <Button variant="fill" color="primary" on:click={toggleOff}>Yes, close this dialog</Button>
+        <Button>No, keep this dialog open</Button>
+      </div>
+    </Dialog>
+  </Toggle>
+</Preview>
+
+<h2>Dispatch closing actions via slot props</h2>
+
+<Preview>
+  <Toggle let:on={open} let:toggle let:toggleOff>
+    <Button on:click={toggle}>Show Dialog</Button>
+    <Dialog
+      {open}
+      on:close={toggleOff}
+      persistent
+      let:attemptClose
+      let:close
+      on:close-attempt={() => alert('attemptClose triggered!')}
+      on:close={() => alert('close triggered!')}
```

Comment:
@techniq `on:close` currently gets triggered twice, which I thus far haven't been able to figure out.

Reply:
Fixed — Resolved in [047cd0a](https://github.com/techniq/svelte-ux/pull/464/commits/047cd0aa4c7e71717f8deeb4427499147593b804)